apps/test/util/testUtils.js

```diff
@@ -118,17 +118,39 @@ exports.runOnStudioTick = function (tick, fn) {
 };
 
 /**
+ * In a bunch of our tests we wrap app.onTick. When we do so, we now also store
+ * a reset function that unwraps it. This method calls all of those functions
+ * at the end of a test to make sure that we're unwrapping everything before
+ * the next test.
+ */
+var wrappersToReset = [];
+exports.resetAppTicks = function () {
+  wrappersToReset.forEach(function (reset) {
+    reset();
+  });
```

Comment:
Do we ever have a test that calls `runOnAppTick` or `tickAppUntil` more than once?  We don't proactively prevent that.  If we do, then order might be important here:

1. Test calls `runOnAppTick` which wraps `onTick` in function `A`
2. Test calls `runOnAppTick` which wraps `A` in function `B`
3. Test calls `resetAppTicks`
4. `A` gets unwrapped, resetting app.onTick to `onTick`
5. `B` gets unwrapped, resetting app.onTick to `A`

I'm not sure it's as simple as iterating backwards either.  Given the same steps 1 and 2, if `A` resolves before `B` it will reset app.onTick to `onTick`, unwrapping `B` even though it hasn't resolved yet.

It might not be an issue now, but I think we should control for it (you almost need tests for this test utility).

It might be better to only ever wrap `onTick` once, in a method that invokes any number of independently registered tick functions that can be independently unregistered.  `Observer`/`ObservableEvent` might be helpful here.